### PR TITLE
feat: Improved App Nav - Top + Inline

### LINKED ISSUE #19160: [Task]: Nav Component - Top + Inline
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

https://www.notion.so/appsmith/Top-Inline-b35e78ffdf0b4efb8335947f3a5872ed

### PR DESCRIPTION
**Pull request by dhruvikn**
## Description

Adds Top + Inline Nav Style

Fixes #19160

## Type of change
- New feature (non-breaking change which adds functionality)

- Manual
- Jest
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/pages/AppViewer/Navigation/TopInline.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useRef, useState } from "react";
+import { useLocation } from "react-router-dom";
+import MenuItemContainer from "./components/MenuItemContainer";
+import MenuItem from "./components/MenuItem";
+import {
+  ApplicationPayload,
+  Page,
+} from "@appsmith/constants/ReduxActionConstants";
+import { NAVIGATION_SETTINGS } from "constants/AppConstants";
+import { get } from "lodash";
```
Comment (jsartisan) [inline app/client/src/pages/AppViewer/Navigation/TopInline.tsx:10]: can we reorder the imports in order of:
1. Third party imports
2. local imports from root
3. imports relative to current folder

This 
```
import React, { useEffect, useRef, useState } from "react";
import { useLocation } from "react-router-dom";
import MenuItemContainer from "./components/MenuItemContainer";
import MenuItem from "./components/MenuItem";
import { getSelectedAppTheme } from "selectors/appThemingSelectors";
```

becomes

```
import { useLocation } from "react-router-dom";
import React, { useEffect, useRef, useState } from "react";

import { getSelectedAppTheme } from "selectors/appThemingSelectors";

import MenuItem from "./components/MenuItem";
import MenuItemContainer from "./components/MenuItemContainer";
```

I also reorder them according to the width :P 
Makes it little more readable.

File: app/client/src/pages/AppViewer/Navigation/components/ApplicationName.styled.tsx [hunk 2]
```diff
@@ -12,5 +12,5 @@ export const StyledApplicationName = styled.div<{
     getApplicationNameTextColor(primaryColor, navColorStyle)};
   font-size: ${({ forSidebar }) =>
     forSidebar ? THEMEING_TEXT_SIZES.sm : THEMEING_TEXT_SIZES.base};
-  max-width: 230px;
+  max-width: 224px;
```
Comment (jsartisan) [inline app/client/src/pages/AppViewer/Navigation/components/ApplicationName.styled.tsx:15]: use a const for this. can't really tell what this magic number is for. ( also i can see its for max-width, but moving to a const makes it more readable )

File: app/client/src/pages/AppViewer/Navigation/TopInline.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  useEffect(() => {
+    if (navRef?.current) {
+      const { offsetWidth } = navRef.current;
+
+      // using max menu item width for simpler calculation
+      setMaxMenuItemsThatCanFit(Math.floor(offsetWidth / maxMenuItemWidth));
+    }
+  }, [navRef, appPages, screenWidth]);
+
+  return appPages.length > 1 ? (
```
Comment (jsartisan) [inline app/client/src/pages/AppViewer/Navigation/TopInline.tsx:68]: I would do:

```
if (!appPages.length === 0) return null;

return 
 <Container className="flex gap-x-2 items-center" ref={navRef}>
      {appPages.map(
        (page, index) =>
          index < maxMenuItemsThatCanFit && (
            <MenuItemContainer
              isTabActive={pathname.indexOf(page.pageId) > -1}
              key={page.pageId}
            >
              <MenuItem
                navigationSetting={currentApplicationDetails?.navigationSetting}
                page={page}
                query={query}
              />
            </MenuItemContainer>
          ),
      )}
````

File: app/client/src/pages/AppViewer/Navigation/components/ApplicationName.styled.tsx (hunk 2 above)
Comment (dhruvikn) [reply app/client/src/pages/AppViewer/Navigation/components/ApplicationName.styled.tsx:15]: 224px was specified by the [design here](https://www.figma.com/file/EkPT7FdqmArkk9L7mySA5S/NAV---Designs?node-id=1983%3A498342&t=lh500UVzK6h031kD-0).

CleanShot 2023-02-07 at 14 03 10@2x

Comment (dhruvikn) [reply app/client/src/pages/AppViewer/Navigation/components/ApplicationName.styled.tsx:15]: Will move it to constants though.

**Review by jsartisan — APPROVED** [review]

**Pull request merged**